__test__/authentication.test.js

```diff
@@ -0,0 +1,110 @@
+const { User } = require('../models');
+const { verifyToken } = require('../helpers/jwt');
+const authentication = require('../middlewares/authentication');
+
+jest.mock('../models', () => ({
+    User: {
+        findOne: jest.fn()
+    }
+}));
+
+jest.mock('../helpers/jwt', () => ({
+    verifyToken: jest.fn()
+}));
+
+const mockRequest = (headers) => ({
+    get: (name) => headers[name]
+});
+
+const mockResponse = () => {
+    const res = {};
+    res.status = jest.fn().mockReturnValue(res);
+    res.json = jest.fn().mockReturnValue(res);
+    res.locals = {};
+    return res;
+};
+
+const mockNext = jest.fn();
+
+describe('Authentication Middleware', () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    test('should call next if user is authenticated', async () => {
+        const userDecoded = { id: 1, email: 'test@example.com' };
+        verifyToken.mockReturnValue(userDecoded);
+        User.findOne.mockResolvedValue({ id: 1, email: 'test@example.com' });
+
+        const req = mockRequest({ token: 'valid_token' });
+        const res = mockResponse();
+        const next = mockNext;
+
+        await authentication(req, res, next);
+
+        expect(verifyToken).toHaveBeenCalledWith('valid_token');
+        expect(User.findOne).toHaveBeenCalledWith({
+            where: { id: userDecoded.id, email: userDecoded.email }
+        });
+        expect(res.locals.user).toEqual({ id: 1, email: 'test@example.com' });
+        expect(next).toHaveBeenCalled();
+    });
+
+    test('should return 401 if user is not found', async () => {
+        const userDecoded = { id: 1, email: 'test@example.com' };
+        verifyToken.mockReturnValue(userDecoded);
+        User.findOne.mockResolvedValue(null);
+
+        const req = mockRequest({ token: 'valid_token' });
+        const res = mockResponse();
+        const next = mockNext;
+
+        await authentication(req, res, next);
+
+        expect(verifyToken).toHaveBeenCalledWith('valid_token');
+        expect(User.findOne).toHaveBeenCalledWith({
+            where: { id: userDecoded.id, email: userDecoded.email }
+        });
+        expect(res.status).toHaveBeenCalledWith(401);
+        expect(res.json).toHaveBeenCalledWith({
+            name: "Authentication Error",
+            devMessage: `User with id ${userDecoded.id} not found in Database`
```

Comment:
ini property untuk apa ya ?

Reply:
property dalam testing ini digunakan untuk simulate kondisi middleware authentication harus return 401 jika user tidak ditemukan di database Pak

Reply:
baik pak, ada kesalahan & fixed the test message